### ISSUE TITLE
 Unable to detect statically whether the number of index_points is 1

### ISSUE THREAD

**Issue by Kuurusch** [tensorflow/probability#838]
Hallo,

When I train a GP with batches I've got the following warning:

> UserWarning: Unable to detect statically whether the number of index_points is 1. As a result, defaulting to treating the marginal GP at `index_points` as a multivariate Gaussian. This makes some methods, like `cdf` unavailable.

Is this a problem or normal when I train with batches?

Following my code:

    batch_date, batch_value = (
            tf.data.Dataset.from_tensor_slices(
                (timestamp_train, train_data.values))
                .shuffle(buffer_size=len(train_data))
                .repeat(count=None)
                .batch(batch_size)
                .make_one_shot_iterator()
                .get_next()
        )

        # Gaussian process with batch data to fit the kernel parameters
        gp_batched = tfd.GaussianProcess(
            mean_fn=mean_fn,
            kernel=kernel,
            index_points=batch_date,
            observation_noise_variance=observation_noise_variance)

**Comment by tirthasheshpatel** [comment]
Your tensor dimensions probably have `None` in them somewhere. This is not a problem during training. The warning is just saying that it was unable to detect if there was only one sample in your batched tensor and a univariate normal was expected in place of a default multivariate normal.

**Comment by csuter** [comment]
@tirthasheshpatel is exactly right. The only difference is that the underlying distribution used will be a 1-d MVN instead of a scalar Normal; there may be a very minor performance impact (hence the warning), but otherwise it's fine. You can try printing the shapes of various things and see if you can find where a `None` shape is coming from.

**Comment by Kuurusch** [comment]
Many thanks for your help!
